### DISCUSSION TITLE
Use mobile number as identifier for front-end login

### DISCUSSION THREAD

**Discussion by prasanna-rkumar** [strapi/strapi#8025] [Legacy Topics]
The [users-permissions](https://github.com/strapi/strapi/tree/master/packages/strapi-plugin-users-permissions) plugin lets you to login with either **username** or **email** as the identifier for front-end login.  
But, my implementation cannot support usernames as a login identifier. Using email is also not possible as I cannot add a unique index to the email field in the admin panel.  
So I added a new field **mobile_number** and I want to use the new field (mobile_number) as the identifier for local auth. How do I do that?

**Comment by lauriejim** [comment]
Hello! This is possible to do that.
You will have to use the **customization concept** to customize the `callback` function of the `users-permissions` plugin.
- Customization concept - https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions
- Function to update - https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-users-permissions/controllers/Auth.js#L21
